en/services/command.md

```diff
@@ -80,12 +80,9 @@ sequenceDiagram;
 The rate at which progress messages are emitted is system-dependent.
 Generally though, the GCS should have a much increased timeout after receiving an ACK with `MAV_RESULT_IN_PROGRESS`.
 
-Typically if a system receives exactly the same command while it is already processing the command, it should continue the operation.
-If the command is a "resend" (confirmation number > 0) the command can be ignored; otherwise the system should respond immediately with a progress update `COMMAND_ACK`.
-
-If a system receives the same command with *different parameters* then in general it must be treated as *new command* (although this should be evaluated based on the specific operation requested):
-So for example:
-- If the new command cancels an operation, the recipient would stop the operation and emit `COMMAND_ACK` with `result=MAV_RESULT_ACCEPTED`
-  > **Note** If another (different) command was used to cancel the operation then it would be ACKd with `MAV_RESULT_ACCEPTED`, and the original command would be ACKd with `MAV_RESULT_FAILED`.
-- If the new command re-starts an operation the recipient should restart the job and emit `result=MAV_RESULT_IN_PROGRESS` etc.
-- If a new command changes some general value (e.g. a target position) this would be a new command.
+Often long running operations may be cancelled, restarted, or updated (continue operation but with new parameters) by sending the same command with different parameters.
+- A cancel command will complete immediately with response `COMMAND_ACK.result=MAV_RESULT_ACCEPTED`.
```

Comment:
I don't think there is general cancel commands. I think this only exists for the mission protocol which doesn't use commands.

Reply:
@julianoes In a previous response you indictated that if the state machine is processing a command and it gets another command it should ack with "temporary rejection".
But we can't have that for a long running command - if you can't cancel it then it simply won't be used. 

So yes, there is no general cancel command, but IMO we have an obligation to state how we recommend you do that. 

This might be done using the same command, in which case I would expect the ACK to be "accepted" and the progress updates to stop.

This might be done with a different command or a message. I'm pretty clear that if you used a special cancel command it would ACK with accepted. What I am not sure about is how the original progress notification update should end. Would it just "cease" having sent its last progress update, or would it still ACK the original command with failed or similar?

Hope this makes sense!

Reply:
I propose that it should ack the original command with "cancelled", so we need to add cancelled.

Also, I think, we need a `COMMAND_CANCEL` message for this. Because it's a reasonable thing to temporarily reject all commands while working on processing a command but listen for `COMMAND_CANCEL` nevertheless.